source/ingress.go

```diff
@@ -37,26 +37,37 @@ import (
 type ingressSource struct {
 	client       kubernetes.Interface
 	namespace    string
-	fqdntemplate *template.Template
+	fqdnTemplate *template.Template
 }
 
-// NewIngressSource creates a new ingressSource with the given client and namespace scope.
-func NewIngressSource(client kubernetes.Interface, namespace string, fqdntemplate string) (Source, error) {
-	var tmpl *template.Template
-	var err error
-	if fqdntemplate != "" {
+// NewIngressSource creates a new ingressSource with the given config.
+func NewIngressSource(cfg *Config) (Source, error) {
+	if cfg.KubeClient == nil {
```

Comment:
IMO this check does not seem very nice. I would prefer to have the old initialiser in-place (which is more intuitive if I want to use `IngressSource` and avoid `store` entirely) and create another one for auto-registration from store: `NewIngressFromConfig`